pkg/network/infraconfigurators/masquerade_test.go

```diff
@@ -124,4 +136,215 @@ var _ = Describe("Masquerade infrastructure configurator", func() {
+
+	handler.EXPECT().NftablesAppendRule(proto, "nat", "postrouting", nftIPString, "saddr", vmIP, "counter", "masquerade").Return(nil)
+	handler.EXPECT().NftablesAppendRule(proto, "nat", "prerouting", "iifname", "eth0", "counter", "jump", "KUBEVIRT_PREINBOUND").Return(nil)
+	handler.EXPECT().NftablesAppendRule(proto, "nat", "postrouting", "oifname", "k6t-eth0", "counter", "jump", "KUBEVIRT_POSTINBOUND").Return(nil)
+
+	for _, chain := range []string{"output", "KUBEVIRT_POSTINBOUND"} {
+		handler.EXPECT().NftablesAppendRule(proto, "nat", chain, "tcp", "dport", fmt.Sprintf("{ %s }", strings.Join(PortsUsedByLiveMigration(), ", ")), nftIPString, "saddr", GetLoopbackAdrress(proto), "counter", "return").Return(nil)
+	}
+
+	handler.EXPECT().NftablesAppendRule(proto, "nat", "KUBEVIRT_PREINBOUND", "counter", "dnat", "to", vmIP).Return(nil)
+	handler.EXPECT().NftablesAppendRule(proto, "nat", "KUBEVIRT_POSTINBOUND", nftIPString, "saddr", fmt.Sprintf("{ %s }", GetLoopbackAdrress(proto)), "counter", "snat", "to", gwIP).Return(nil)
+	handler.EXPECT().NftablesAppendRule(proto, "nat", "output", nftIPString, "daddr", fmt.Sprintf("{ %s }", GetLoopbackAdrress(proto)), "counter", "dnat", "to", vmIP).Return(nil)
+}
+
+func mockIPTablesBackend(handler *netdriver.MockNetworkHandler, proto iptables.Protocol, nftIPString string, vmIP string, gwIP string) {
+	handler.EXPECT().GetNFTIPString(proto).Return(nftIPString).AnyTimes()
+	handler.EXPECT().IptablesNewChain(proto, "nat", "KUBEVIRT_PREINBOUND").Return(nil)
+	handler.EXPECT().IptablesNewChain(proto, "nat", "KUBEVIRT_POSTINBOUND").Return(nil)
+
+	handler.EXPECT().IptablesAppendRule(proto, "nat",
+		"POSTROUTING",
+		"-s",
+		vmIP,
+		"-j",
+		"MASQUERADE").Return(nil)
+	handler.EXPECT().IptablesAppendRule(proto, "nat",
+		"PREROUTING",
+		"-i",
+		"eth0",
+		"-j",
+		"KUBEVIRT_PREINBOUND").Return(nil)
+	handler.EXPECT().IptablesAppendRule(proto, "nat",
+		"POSTROUTING",
+		"-o",
+		"k6t-eth0",
+		"-j",
+		"KUBEVIRT_POSTINBOUND").Return(nil)
+	handler.EXPECT().IptablesAppendRule(proto, "nat",
+		"KUBEVIRT_PREINBOUND",
+		"-j",
+		"DNAT",
+		"--to-destination",
+		vmIP).Return(nil)
+	handler.EXPECT().IptablesAppendRule(proto, "nat",
+		"KUBEVIRT_POSTINBOUND",
+		"--source",
+		GetLoopbackAdrress(proto),
+		"-j",
+		"SNAT",
+		"--to-source",
+		gwIP).Return(nil)
+	handler.EXPECT().IptablesAppendRule(proto, "nat",
+		"OUTPUT",
+		"--destination",
+		GetLoopbackAdrress(proto),
+		"-j",
+		"DNAT",
+		"--to-destination",
+		vmIP).Return(nil)
+
+	for _, chain := range []string{"OUTPUT", "KUBEVIRT_POSTINBOUND"} {
+		handler.EXPECT().IptablesAppendRule(proto, "nat", chain,
+			"-p", "tcp", "--match", "multiport",
+			"--dports", fmt.Sprintf("%s", strings.Join(PortsUsedByLiveMigration(), ",")),
+			"--source", GetLoopbackAdrress(proto), "-j", "RETURN").Return(nil)
+	}
+}
+
+func getProtocols(optionalIPProtocol ...iptables.Protocol) []iptables.Protocol {
```

Comment:
nit: It is common in Golang not to use `get*` prefixes. Calling this `protocols` is nicer.